File: output/GPT4o-Mini/xCodeEval_4000/C++/C/980770e7a017f49149b45e925e6fa9d6.c
#include <stdio.h>
#include <math.h>
#include <limits.h>

#define checkMin(a, b) if (b < a) a = b;
#define checkMax(a, b) if (b > a) a = b;

int main() {
    long long a = -1, b = -1;
    int n;

    scanf("%d", &n);

    int tttt = sqrt(n);

    for (int i = 1; i <= tttt + 8; ++i) {
        if (n % i == 0) {
            int t = n / i;
            int zt = sqrt(t);

            for (int j = 1; j <= zt + 8; ++j) {
                if (t % j == 0) {
                    long long tt = t / j;
                    if (tt >= 1) {
                        long long ttt = (tt + 2) * (i + 1) * (j + 2);
                        if (a == -1) a = ttt;
                        else checkMin(a, ttt);
                        if (b == -1) b = ttt;
                        else checkMax(b, ttt);
                    }
                }
            }

            int zz = i;
            i = n / i;

            t = n / i;
            zt = sqrt(t);

            for (int j = 1; j <= zt + 8; ++j) {
                if (t % j == 0) {
                    long long tt = t / j;
                    if (tt >= 1) {
                        long long ttt = (tt + 2) * (i + 1) * (j + 2);
                        if (a == -1) a = ttt;
                        else checkMin(a, ttt);
                        if (b == -1) b = ttt;
                        else checkMax(b, ttt);
                    }
                }
            }

            i = zz;
        }
    }

    printf("%lld %lld\n", a - n, b - n);
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Python/980770e7a017f49149b45e925e6fa9d6.py
import math

def check_min(a, b):
    if b < a:
        return b
    return a

def check_max(a, b):
    if b > a:
        return b
    return a

n = int(input())

a = -1
b = -1

tttt = int(math.sqrt(n))

for i in range(1, tttt + 10):
    if n % i == 0:
        t = n // i
        zt = int(math.sqrt(t))

        for j in range(1, zt + 10):
            if t % j == 0:
                tt = t // j
                if tt >= 1:
                    ttt = (tt + 2) * (i + 1) * (j + 2)

                    if a == -1:
                        a = ttt
                    else:
                        a = check_min(a, ttt)

                    if b == -1:
                        b = ttt
                    else:
                        b = check_max(b, ttt)

        zz = i
        i = n // i

        t = n // i
        zt = int(math.sqrt(t))

        for j in range(1, zt + 10):
            if t % j == 0:
                tt = t // j
                if tt >= 1:
                    ttt = (tt + 2) * (i + 1) * (j + 2)

                    if a == -1:
                        a = ttt
                    else:
                        a = check_min(a, ttt)

                    if b == -1:
                        b = ttt
                    else:
                        b = check_max(b, ttt)

        i = zz

print(a - n, b - n)
